tests/modeling/test_roi_pooler.py

```diff
@@ -134,6 +138,89 @@ def test_no_images(self):
 
+    @unittest.skipIf(TORCH_VERSION < (1, 6), "Insufficient pytorch version")
+    def test_fmt_box_list_onnx_export(self):
+        class Model(torch.nn.Module):
+            def forward(self, box_tensor):
+                return _fmt_box_list(box_tensor, 0)
+
+        with tempfile.TemporaryFile("w+b") as f:
+            torch.onnx.export(
+                Model(),
+                torch.ones(10, 4),
+                f,
+                input_names=["boxes"],
+                output_names=["formatted_boxes"],
+                dynamic_axes={"boxes": {0: "box_count"}, "formatted_boxes": {0: "box_count"}},
+                opset_version=11,
+            )
+
+            f.seek(0)
+
+            sess = rt.InferenceSession(f.read(), None)
+
+            assert sess.run([], {"boxes": np.ones((10, 4), dtype=np.float32)})[0].shape == (10, 5)
+            assert sess.run([], {"boxes": np.ones((5, 4), dtype=np.float32)})[0].shape == (5, 5)
+            assert sess.run([], {"boxes": np.ones((20, 4), dtype=np.float32)})[0].shape == (20, 5)
+
+    @unittest.skipIf(TORCH_VERSION < (1, 6), "Insufficient pytorch version")
+    def test_roi_pooler_onnx_export(self):
+        class Model(torch.nn.Module):
+            def __init__(self, roi):
+                super(Model, self).__init__()
+                self.roi = roi
+
+            def forward(self, x, boxes):
+                return self.roi([x], [Boxes(boxes)])
+
+        pooler_resolution = 14
+        canonical_level = 4
+        canonical_scale_factor = 2 ** canonical_level
+        pooler_scales = (1.0 / canonical_scale_factor,)
+        sampling_ratio = 0
+
+        N, C, H, W = 1, 4, 10, 8
+        N_rois = 10
+        std = 11
+        mean = 0
+        feature = (torch.rand(N, C, H, W) - 0.5) * 2 * std + mean
+
+        rois = self._rand_boxes(
+            num_boxes=N_rois, x_max=W * canonical_scale_factor, y_max=H * canonical_scale_factor
+        )
+
+        model = Model(
+            ROIPooler(
+                output_size=pooler_resolution,
+                scales=pooler_scales,
+                sampling_ratio=sampling_ratio,
+                pooler_type="ROIAlign",
+            )
+        )
+
+        with tempfile.TemporaryFile("w+b") as f:
+            torch.onnx.export(
+                model,
+                (feature, rois),
+                f,
+                input_names=["features", "boxes"],
+                output_names=["pooled_features"],
+                dynamic_axes={"boxes": {0: "detections"}, "pooled_features": {0: "detections"}},
+                opset_version=11,
+            )
+
+            f.seek(0)
+
+            sess = rt.InferenceSession(f.read(), None)
+
+            assert (
+                sess.run([], {"features": feature.numpy(), "boxes": rois.numpy()})[0].shape[0] == 10
+            )
+            assert (
+                sess.run([], {"features": feature.numpy(), "boxes": rois[:5].numpy()})[0].shape[0]
+                == 5
+            )
```

Comment:
```suggestion
        func = torch.jit.trace(model, (feature, rois))
        o = func(feature, rois)
        self.assertEqual(o.shape, (10, 4, 14, 14))
        o = func(feature, rois[:5])
        self.assertEqual(o.shape, (5, 4, 14, 14))

```

I think this change will test just the same feature you're trying to support. 
Please don't rely on onnxruntime in this PR then. It isn't very friendly to maintainers given it now requires libomp. (also we in general can't provide much support for third-party runtime)

Reply:
Thank you for the suggestion. That works. Though the scripting does seem to run into [this bug](https://github.com/pytorch/pytorch/pull/44335) while applying the box deltas in the box regressor.